packages/ag-charts-community/src/chart/legend/legend.ts

```diff
@@ -302,13 +303,10 @@ export class Legend extends BaseProperties {
         const visibleItems = legendData.filter((datum) => !datum.hideInLegend);
         const { data } = this;
 
-        const propertiesToCheck: (keyof CategoryLegendDatum)[] = ['seriesId', 'itemId', 'enabled', 'legendItemName'];
+        const propertiesToCheck = ['seriesId', 'itemId', 'enabled', 'legendItemName', 'label.text'];
         const shouldUpdate =
             data.length !== legendData.length ||
-            data.some((_v, index, _a) => {
-                const [newValue, oldValue] = [legendData[index], data[index]];
-                return propertiesToCheck.some((p) => newValue[p] !== oldValue[p]);
-            });
+            data.some((_v, index, _a) => !objectsEqualKeys(propertiesToCheck, legendData[index], data[index]));
```

Comment:
It might be we can just do the complete `objectsEqual()` check here, not sure if there is reason we are comparing specific properties? @manapeirov 

But this just adds `'label.text'` to the list to compare.

Reply:
Yeah we can just do objectsEqual() technically everything should be the same for us not to update it

Reply:
Updated to use `objectsEqual()`.